### PR TITLE
[qwt] adding qwt library

### PR DESCRIPTION
**Pull request by ig-or**
+ qwt library for Windows and Linux x64

### CODE REVIEW

File: Q/qwt/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+cp $prefix/lib/qwtd.dll $prefix/bin/qwtd.dll
+cp $prefix/lib/qwtd.dll $prefix/bin/libqwtd.dll
+cd $WORKSPACE/srcdir
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Windows(:x86_64),
+    Linux(:x86_64, libc=:glibc)
```
Comment (giordano) [inline Q/qwt/build_tarballs.jl:31]: Sorry, here we're using a slightly different syntax for the platforms
```suggestion
    Platform("x86_64", "windows"),
    Platform("x86_64", "linux"; libc="glibc"),
```

File: Q/qwt/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+cd $WORKSPACE/srcdir
+cd qwt
+qmake
+make -j 
+make install
+cp $prefix/lib/qwt.dll $prefix/bin/qwt.dll
+cp $prefix/lib/qwt.dll $prefix/bin/libqwt.dll
+cp $prefix/lib/qwtd.dll $prefix/bin/qwtd.dll
+cp $prefix/lib/qwtd.dll $prefix/bin/libqwtd.dll
+cd $WORKSPACE/srcdir
```
Comment (giordano) [inline Q/qwt/build_tarballs.jl:24]: I think the libraries will automatically be moved, so we can simplify the build script quite a bit:
```suggestion
cd $WORKSPACE/srcdir/qwt
qmake
make -j${nproc}
make install
```

File: Q/qwt/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+# platforms are passed in on the command line
+platforms = [
+    Platform("x86_64", "windows"),
+    Platform("x86_64", "linux"; libc="glibc"),
+]
+
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libqwt", :qwt)
```
Comment (giordano) [inline Q/qwt/build_tarballs.jl:37]:
```suggestion
    LibraryProduct(["libqwt", "qwt"], :qwt),
```

File: Q/qwt/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="Qt_jll", uuid="ede63266-ebff-546c-83e0-1c6fb6d0efc8"))
+    Dependency(PackageSpec(name="GLEW_jll", uuid="bde7f898-03f7-559e-8810-194d950ce600"))
+    Dependency(PackageSpec(name="GLFW_jll", uuid="0656b61e-2033-5cc2-a64a-77c0f6c09b89"))
+    Dependency(PackageSpec(name="GLU_jll", uuid="bd17208b-e95e-5925-bf81-e2f59b3e5c61"))
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"9.1.0")
```
Comment (giordano) [inline Q/qwt/build_tarballs.jl:43]: Why are you using GCC 9?

Comment (ig-or) [reply Q/qwt/build_tarballs.jl:43]: Maybe we can set it back to the default value? anything with C++14 support will work.  did not believed it is critical..

Comment (giordano) [reply Q/qwt/build_tarballs.jl:43]: Ok, GCC v5 should be enough
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"5")
```

**Comment by giordano** [comment]
Thank you!

**Comment by ig-or** [comment]
Thank you very much! You are so fast

**Pull request merged**